homeassistant/components/mqtt/climate.py

```diff
@@ -668,28 +673,41 @@ def config_schema() -> vol.Schema:
     def _setup_from_config(self, config: ConfigType) -> None:
         """(Re)Setup the entity."""
         self._attr_hvac_modes = config[CONF_MODE_LIST]
-        self._attr_min_temp = config[CONF_TEMP_MIN]
-        self._attr_max_temp = config[CONF_TEMP_MAX]
+        # Make sure the min an max temp is converted to the correct when not set
+        self._attr_temperature_unit = config.get(
+            CONF_TEMPERATURE_UNIT, self.hass.config.units.temperature_unit
+        )
+        if (min_temp := config.get(CONF_TEMP_MIN)) is not None:
+            self._attr_min_temp = min_temp
+        if (max_temp := config.get(CONF_TEMP_MAX)) is not None:
+            self._attr_max_temp = max_temp
         self._attr_min_humidity = config[CONF_HUMIDITY_MIN]
         self._attr_max_humidity = config[CONF_HUMIDITY_MAX]
-        self._attr_precision = config.get(CONF_PRECISION, super().precision)
+        if (precision := config.get(CONF_PRECISION)) is not None:
+            self._attr_precision = precision
```

Comment:
As with `max_temp` and `min_temp` we should no set the attribute to ensure a suggested precision when the temperature unit is changed.